src/main/java/org/reactome/addlinks/referencecreators/KEGGReferenceCreator.java

```diff
@@ -95,48 +95,51 @@ public void createIdentifiers(long personID, Map<String, List<Map<KEGGKeys, Stri
 					{
 						keggPrefix = KEGGSpeciesCache.extractKEGGSpeciesCode(keggGeneIdentifier);
 					}
-
-//					// If the original KEGG_IDENTIFIER key didn't have a value, use the KEGG GENE ID.
-					if (keggIdentifier == null || keggIdentifier.trim().equals("") )
+					// Only proceed if the prefix is not forbidden (forbidden prefixes are set in the reference-creators.xml file).
+					if (!KEGGReferenceCreatorHelper.isKEGGPrefixForbidden(keggPrefix))
 					{
-						keggIdentifier = keggGeneIdentifier;
-					}
-
-					logger.trace("Working on source object: {}", sourceReference.toString());
-					String targetDB = null;
-					KEGGReferenceCreatorHelper referenceCreatorHelper = new KEGGReferenceCreatorHelper(objectCache, this.logger);
-					String[] parts = referenceCreatorHelper.determineKeggReferenceDatabase(keggGeneIdentifier, keggPrefix);
-					targetDB = parts[0];
-					// now, the identifier should have been cleaned up by determineKeggReferenceDatabase
-					keggGeneIdentifier = parts[1];
-					xrefAlreadyExists = this.checkXRefExists(sourceReference, keggGeneIdentifier, targetDB);
-					if (!xrefAlreadyExists)
-					{
-						String keggDefinition = keggData.get(KEGGKeys.KEGG_DEFINITION);
-						sourceIdentifiersWithNewIdentifier++;
-						// Also need to add the keggDefinition and keggGeneID as "name" attributes.
-						List<String> names = new ArrayList<String>(3);
-						names.add(keggDefinition);
-						names.add(keggGeneIdentifier);
-						keggIdentifier = KEGGSpeciesCache.pruneKEGGSpeciesCode(keggIdentifier);
-						if (! keggIdentifier.equals(keggGeneIdentifier))
+//						// If the original KEGG_IDENTIFIER key didn't have a value, use the KEGG GENE ID.
+						if (keggIdentifier == null || keggIdentifier.trim().equals("") )
 						{
-							names.add(keggIdentifier);
+							keggIdentifier = keggGeneIdentifier;
 						}
-						Map<String,List<String>> extraAttributes = new HashMap<String,List<String>>(1);
-						extraAttributes.put(ReactomeJavaConstants.name, names);
-						logger.trace("For {}, creating new KEGG xref: {}",sourceReference.getDisplayName(), keggIdentifier);
-						if (!this.testMode)
+
+						logger.trace("Working on source object: {}", sourceReference.toString());
+						String targetDB = null;
+						KEGGReferenceCreatorHelper referenceCreatorHelper = new KEGGReferenceCreatorHelper(objectCache, logger);
```

Comment:
Should `logger` be referenced as `this.logger` here and on line 107 for consistency with the rest of the codebase?

Reply:
Hmm... possibly...

Reply:
It's consistent within the class. Earlier, I fell out of the habit of qualifying loggers with `this`.

Reply:
Yes, the codebase is not 100% consistent, but I'm more focused on fixing the bug here, then making everything consistent.

Reply:
Okay, no problem.  I just noticed the difference compared to other parts of the PR.